src/status_im/ethereum/subscriptions.cljs

```diff
@@ -37,22 +79,117 @@
 
 (defn subscribe-signal
   [filter params callback]
-  (status/call-private-rpc
-   (.stringify js/JSON (clj->js {:jsonrpc "2.0"
-                                 :id      1
-                                 :method  "eth_subscribeSignal"
-                                 :params  [filter, params]}))
-   (fn [response]
-     (if (string/blank? response)
-       (log/error ::subscription-unknown-error :filter filter :params params)
-       (let [{:keys [error result]}
-             (-> (.parse js/JSON response)
-                 (js->clj :keywordize-keys true))]
-         (if error
-           (log/error ::subscription-error error :filter filter :params params)
-           (re-frame/dispatch [:ethereum.callback/subscription-success
-                               result
-                               callback])))))))
+  (let [args (.stringify js/JSON (clj->js {:jsonrpc "2.0"
+                                           :id      1
+                                           :method  "eth_subscribeSignal"
+                                           :params  [filter params]}))]
+    (status/call-private-rpc
+     args
+     (fn [response]
+       (if (string/blank? response)
+         (log/error ::subscription-unknown-error :filter filter :params params)
+         (let [{:keys [error result]}
+               (-> (.parse js/JSON response)
+                   (js->clj :keywordize-keys true))]
+           (if error
+             (log/error ::subscription-error error :filter filter :params params)
+             (re-frame/dispatch [:ethereum.callback/subscription-success
+                                 result
+                                 callback]))))))))
+
+(defn- add-padding [address]
+  {:pre [(string? address)]}
+  (str "0x000000000000000000000000" (subs address 2)))
+
+(defn- remove-padding [topic]
+  {:pre [(string? topic)]}
+  (str "0x" (subs topic 26)))
+
+(defn- parse-transaction-entries [timestamp chain-tokens direction transfers]
+  {:pre [(integer? timestamp)
+         (map? chain-tokens)
+         (every? (fn [[k v]] (and (string? k) (map? v))) chain-tokens)
+         (keyword? direction)
+         (every? map? transfers)]}
+  (into {}
+        (keep identity
+              (for [transfer transfers]
+                (when-let [token (->> transfer :address (get chain-tokens))]
```

Comment:
Maybe makes sense to extract into a separate fn.

Reply:
it will be replaced in next PR anyway